### PR TITLE
Add the Abats Keyboard Club NewJeans keyboard

### PR DESCRIPTION
**Pull request by ramonimbao**
## Description

Adds the Abats Keyboard Club NewJeans keyboard. Has a few VIA custom menus used for controlling the badge LEDs. Contains both solder and hotswap variants.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/abatskeyboardclub/newjeans/solder/info.json [hunk 1]
```diff
@@ -0,0 +1,956 @@
...
+            {"flags": 4, "x": 35, "y":64},
+            {"flags": 4, "x": 90, "y":64},
+            {"flags": 4, "x":144, "y":64},
+            {"flags": 4, "x":160, "y":64},
+            {"flags": 4, "x":176, "y":64},
+            {"flags": 4, "x":195, "y":64},
+            {"flags": 4, "x":208, "y":64},
+            {"flags": 4, "x":221, "y":64}
+        ]
+    },
```
Comment (dunk2k) [inline keyboards/abatskeyboardclub/newjeans/solder/info.json:177]:
```suggestion
    },
    "community_layouts": ["tkl_ansi_tsangan", "tkl_ansi_tsangan_split_bs_rshift", "tkl_ansi_f13_tsangan", "tkl_ansi_f13_tsangan_split_bs_rshift", "tkl_iso_f13_tsangan", "tkl_iso_f13_tsangan_split_bs_rshift", "tkl_iso_tsangan", "tkl_iso_tsangan_split_bs_rshift" ],
```

File: keyboards/abatskeyboardclub/newjeans/solder/info.json [hunk 2]
```diff
@@ -0,0 +1,956 @@
...
+    "layouts": {
+        "LAYOUT_tkl_f12_ansi": {
```
Comment (dunk2k) [inline keyboards/abatskeyboardclub/newjeans/solder/info.json:179]:
```suggestion
        "LAYOUT_tkl_ansi_tsangan": {
```

File: keyboards/abatskeyboardclub/newjeans/solder/info.json [hunk 3]
```diff
@@ -0,0 +1,956 @@
...
+                {"matrix": [5, 7], "x": 4, "y": 5.25, "w": 7},
+                {"matrix": [5, 11], "x": 11, "y": 5.25, "w": 1.5},
+                {"matrix": [5, 12], "x": 12.5, "y": 5.25},
+                {"matrix": [5, 13], "x": 13.5, "y": 5.25, "w": 1.5},
+                {"matrix": [5, 14], "x": 15.25, "y": 5.25},
+                {"matrix": [5, 15], "x": 16.25, "y": 5.25},
+                {"matrix": [5, 16], "x": 17.25, "y": 5.25}
+            ]
+        },
+        "LAYOUT_tkl_f12_ansi_split_bs_rshift": {
```
Comment (dunk2k) [inline keyboards/abatskeyboardclub/newjeans/solder/info.json:274]:
```suggestion
        "LAYOUT_tkl_ansi_tsangan_split_bs_rshift": {
```

File: keyboards/abatskeyboardclub/newjeans/solder/info.json [hunk 4]
```diff
@@ -0,0 +1,956 @@
...
+                {"matrix": [5, 7], "x": 4, "y": 5.25, "w": 7},
+                {"matrix": [5, 11], "x": 11, "y": 5.25, "w": 1.5},
+                {"matrix": [5, 12], "x": 12.5, "y": 5.25},
+                {"matrix": [5, 13], "x": 13.5, "y": 5.25, "w": 1.5},
+                {"matrix": [5, 14], "x": 15.25, "y": 5.25},
+                {"matrix": [5, 15], "x": 16.25, "y": 5.25},
+                {"matrix": [5, 16], "x": 17.25, "y": 5.25}
+            ]
+        },
+        "LAYOUT_tkl_f13_ansi": {
```
Comment (dunk2k) [inline keyboards/abatskeyboardclub/newjeans/solder/info.json:371]:
```suggestion
        "LAYOUT_tkl_f13_ansi_tsangan": {
```

File: keyboards/abatskeyboardclub/newjeans/solder/info.json [hunk 5]
```diff
@@ -0,0 +1,956 @@
...
+                {"matrix": [5, 7], "x": 4, "y": 5.25, "w": 7},
+                {"matrix": [5, 11], "x": 11, "y": 5.25, "w": 1.5},
+                {"matrix": [5, 12], "x": 12.5, "y": 5.25},
+                {"matrix": [5, 13], "x": 13.5, "y": 5.25, "w": 1.5},
+                {"matrix": [5, 14], "x": 15.25, "y": 5.25},
+                {"matrix": [5, 15], "x": 16.25, "y": 5.25},
+                {"matrix": [5, 16], "x": 17.25, "y": 5.25}
+            ]
+        },
+        "LAYOUT_tkl_f13_ansi_split_bs_rshift": {
```
Comment (dunk2k) [inline keyboards/abatskeyboardclub/newjeans/solder/info.json:467]:
```suggestion
        "LAYOUT_tkl_f13_ansi_tsangan_split_bs_rshift": {
```

File: keyboards/abatskeyboardclub/newjeans/solder/info.json [hunk 6]
```diff
@@ -0,0 +1,956 @@
...
+                {"matrix": [5, 11], "x": 11, "y": 5.25, "w": 1.5},
+                {"matrix": [5, 12], "x": 12.5, "y": 5.25},
+                {"matrix": [5, 13], "x": 13.5, "y": 5.25, "w": 1.5},
+                {"matrix": [5, 14], "x": 15.25, "y": 5.25},
+                {"matrix": [5, 15], "x": 16.25, "y": 5.25},
+                {"matrix": [5, 16], "x": 17.25, "y": 5.25}
+            ]
+        },
+
+        "LAYOUT_tkl_f12_iso": {
```
Comment (dunk2k) [inline keyboards/abatskeyboardclub/newjeans/solder/info.json:566]:
```suggestion
        "LAYOUT_tkl_iso_tsangan": {
```

File: keyboards/abatskeyboardclub/newjeans/solder/info.json [hunk 7]
```diff
@@ -0,0 +1,956 @@
...
+                {"matrix": [5, 7], "x": 4, "y": 5.25, "w": 7},
+                {"matrix": [5, 11], "x": 11, "y": 5.25, "w": 1.5},
+                {"matrix": [5, 12], "x": 12.5, "y": 5.25},
+                {"matrix": [5, 13], "x": 13.5, "y": 5.25, "w": 1.5},
+                {"matrix": [5, 14], "x": 15.25, "y": 5.25},
+                {"matrix": [5, 15], "x": 16.25, "y": 5.25},
+                {"matrix": [5, 16], "x": 17.25, "y": 5.25}
+            ]
+        },
+        "LAYOUT_tkl_f12_iso_split_bs_rshift": {
```
Comment (dunk2k) [inline keyboards/abatskeyboardclub/newjeans/solder/info.json:661]:
```suggestion
        "LAYOUT_tkl_iso_tsangan_split_bs_rshift": {
```

File: keyboards/abatskeyboardclub/newjeans/solder/info.json [hunk 8]
```diff
@@ -0,0 +1,956 @@
...
+                {"matrix": [5, 7], "x": 4, "y": 5.25, "w": 7},
+                {"matrix": [5, 11], "x": 11, "y": 5.25, "w": 1.5},
+                {"matrix": [5, 12], "x": 12.5, "y": 5.25},
+                {"matrix": [5, 13], "x": 13.5, "y": 5.25, "w": 1.5},
+                {"matrix": [5, 14], "x": 15.25, "y": 5.25},
+                {"matrix": [5, 15], "x": 16.25, "y": 5.25},
+                {"matrix": [5, 16], "x": 17.25, "y": 5.25}
+            ]
+        },
+        "LAYOUT_tkl_f13_iso": {
```
Comment (dunk2k) [inline keyboards/abatskeyboardclub/newjeans/solder/info.json:759]:
```suggestion
        "LAYOUT_tkl_f13_iso_tsangan": {
```

File: keyboards/abatskeyboardclub/newjeans/solder/info.json [hunk 9]
```diff
@@ -0,0 +1,956 @@
...
+                {"matrix": [5, 7], "x": 4, "y": 5.25, "w": 7},
+                {"matrix": [5, 11], "x": 11, "y": 5.25, "w": 1.5},
+                {"matrix": [5, 12], "x": 12.5, "y": 5.25},
+                {"matrix": [5, 13], "x": 13.5, "y": 5.25, "w": 1.5},
+                {"matrix": [5, 14], "x": 15.25, "y": 5.25},
+                {"matrix": [5, 15], "x": 16.25, "y": 5.25},
+                {"matrix": [5, 16], "x": 17.25, "y": 5.25}
+            ]
+        },
+        "LAYOUT_tkl_f13_iso_split_bs_rshift": {
```
Comment (dunk2k) [inline keyboards/abatskeyboardclub/newjeans/solder/info.json:856]:
```suggestion
        "LAYOUT_tkl_f13_iso_tsangan_split_bs_rshift": {
```

File: keyboards/abatskeyboardclub/newjeans/hotswap/config.h [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+/* disable debug print */
+//#define NO_DEBUG
+
+/* disable print */
+//#define NO_PRINT
+
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
```
Comment (dunk2k) [inline keyboards/abatskeyboardclub/newjeans/hotswap/config.h:33]:
```suggestion
```

**Review by dunk2k — CHANGES_REQUESTED** [review]
1. add Community Layout support
    1. rename layout names to conform 
2. remove unnecessary RGB definitions in `config.h`

File: keyboards/abatskeyboardclub/newjeans/solder/info.json [hunk 11]
```diff
@@ -0,0 +1,956 @@
...
+                {"matrix": [5, 2], "x": 2.5, "y": 5.25, "w": 1.5},
+                {"matrix": [5, 7], "x": 4, "y": 5.25, "w": 7},
+                {"matrix": [5, 11], "x": 11, "y": 5.25, "w": 1.5},
+                {"matrix": [5, 12], "x": 12.5, "y": 5.25},
+                {"matrix": [5, 13], "x": 13.5, "y": 5.25, "w": 1.5},
+                {"matrix": [5, 14], "x": 15.25, "y": 5.25},
+                {"matrix": [5, 15], "x": 16.25, "y": 5.25},
+                {"matrix": [5, 16], "x": 17.25, "y": 5.25}
+            ]
+        }
```
Comment (dunk2k) [inline keyboards/abatskeyboardclub/newjeans/solder/info.json:954]:
```suggestion
        },
        "LAYOUT_all": {
            [
                {"matrix": [0, 0], "x": 0, "y": 0},
                {"matrix": [0, 1], "x": 1.25, "y": 0},
                {"matrix": [0, 2], "x": 2.25, "y": 0},
                {"matrix": [0, 3], "x": 3.25, "y": 0},
                {"matrix": [0, 4], "x": 4.25, "y": 0},
                {"matrix": [0, 5], "x": 5.5, "y": 0},
                {"matrix": [0, 6], "x": 6.5, "y": 0},
                {"matrix": [0, 7], "x": 7.5, "y": 0},
                {"matrix": [0, 8], "x": 8.5, "y": 0},
                {"matrix": [0, 9], "x": 9.75, "y": 0},
                {"matrix": [0, 10], "x": 10.75, "y": 0},
                {"matrix": [0, 11], "x": 11.75, "y": 0},
                {"matrix": [0, 12], "x": 12.75, "y": 0},
                {"matrix": [0, 13], "x": 14, "y": 0},
                {"matrix": [0, 14], "x": 15.25, "y": 0},
                {"matrix": [0, 15], "x": 16.25, "y": 0},
                {"matrix": [0, 16], "x": 17.25, "y": 0},

                {"matrix": [1, 0], "x": 0, "y": 1.25},
                {"matrix": [1, 1], "x": 1, "y": 1.25},
                {"matrix": [1, 2], "x": 2, "y": 1.25},
                {"matrix": [1, 3], "x": 3, "y": 1.25},
                {"matrix": [1, 4], "x": 4, "y": 1.25},
                {"matrix": [1, 5], "x": 5, "y": 1.25},
                {"matrix": [1, 6], "x": 6, "y": 1.25},
                {"matrix": [1, 7], "x": 7, "y": 1.25},
                {"matrix": [1, 8], "x": 8, "y": 1.25},
                {"matrix": [1, 9], "x": 9, "y": 1.25},
                {"matrix": [1, 10], "x": 10, "y": 1.25},
                {"matrix": [1, 11], "x": 11, "y": 1.25},
                {"matrix": [1, 12], "x": 12, "y": 1.25},
                {"matrix": [1, 13], "x": 13, "y": 1.25},
                {"matrix": [3, 14], "x": 14, "y": 1.25},
                {"matrix": [1, 14], "x": 15.25, "y": 1.25},
                {"matrix": [1, 15], "x": 16.25, "y": 1.25},
                {"matrix": [1, 16], "x": 17.25, "y": 1.25},

                {"matrix": [2, 0], "x": 0, "y": 2.25, "w": 1.5},
                {"matrix": [2, 1], "x": 1.5, "y": 2.25},
                {"matrix": [2, 2], "x": 2.5, "y": 2.25},
                {"matrix": [2, 3], "x": 3.5, "y": 2.25},
                {"matrix": [2, 4], "x": 4.5, "y": 2.25},
                {"matrix": [2, 5], "x": 5.5, "y": 2.25},
                {"matrix": [2, 6], "x": 6.5, "y": 2.25},
                {"matrix": [2, 7], "x": 7.5, "y": 2.25},
                {"matrix": [2, 8], "x": 8.5, "y": 2.25},
                {"matrix": [2, 9], "x": 9.5, "y": 2.25},
                {"matrix": [2, 10], "x": 10.5, "y": 2.25},
                {"matrix": [2, 11], "x": 11.5, "y": 2.25},
                {"matrix": [2, 12], "x": 12.5, "y": 2.25},
                {"matrix": [2, 13], "x": 13.5, "y": 2.25, "w": 1.5},
                {"matrix": [2, 14], "x": 15.25, "y": 2.25},
                {"matrix": [2, 15], "x": 16.25, "y": 2.25},
                {"matrix": [2, 16], "x": 17.25, "y": 2.25},

                {"matrix": [3, 0], "x": 0, "y": 3.25, "w": 1.75},
                {"matrix": [3, 1], "x": 1.75, "y": 3.25},
                {"matrix": [3, 2], "x": 2.75, "y": 3.25},
                {"matrix": [3, 3], "x": 3.75, "y": 3.25},
                {"matrix": [3, 4], "x": 4.75, "y": 3.25},
                {"matrix": [3, 5], "x": 5.75, "y": 3.25},
                {"matrix": [3, 6], "x": 6.75, "y": 3.25},
                {"matrix": [3, 7], "x": 7.75, "y": 3.25},
                {"matrix": [3, 8], "x": 8.75, "y": 3.25},
                {"matrix": [3, 9], "x": 9.75, "y": 3.25},
                {"matrix": [3, 10], "x": 10.75, "y": 3.25},
                {"matrix": [3, 11], "x": 11.75, "y": 3.25},
                {"matrix": [3, 12], "x": 12.75, "y": 3.25},
                {"matrix": [3, 13], "x": 13.75, "y": 3.25, "w": 1.25},

                {"matrix": [4, 0], "x": 0, "y": 4.25, "w": 2.25},
                {"matrix": [4, 2], "x": 2.25, "y": 4.25},
                {"matrix": [4, 3], "x": 3.25, "y": 4.25},
                {"matrix": [4, 4], "x": 4.25, "y": 4.25},
                {"matrix": [4, 5], "x": 5.25, "y": 4.25},
                {"matrix": [4, 6], "x": 6.25, "y": 4.25},
                {"matrix": [4, 7], "x": 7.25, "y": 4.25},
                {"matrix": [4, 8], "x": 8.25, "y": 4.25},
                {"matrix": [4, 9], "x": 9.25, "y": 4.25},
                {"matrix": [4, 10], "x": 10.25, "y": 4.25},
                {"matrix": [4, 11], "x": 11.25, "y": 4.25},
                {"matrix": [4, 12], "x": 12.25, "y": 4.25, "w": 1.75},
                {"matrix": [4, 13], "x": 14, "y": 4.25},
                {"matrix": [4, 15], "x": 16.25, "y": 4.25},

                {"matrix": [5, 0], "x": 0, "y": 5.25, "w": 1.5},
                {"matrix": [5, 1], "x": 1.5, "y": 5.25},
                {"matrix": [5, 2], "x": 2.5, "y": 5.25, "w": 1.5},
                {"matrix": [5, 7], "x": 4, "y": 5.25, "w": 7},
                {"matrix": [5, 11], "x": 11, "y": 5.25, "w": 1.5},
                {"matrix": [5, 12], "x": 12.5, "y": 5.25},
                {"matrix": [5, 13], "x": 13.5, "y": 5.25, "w": 1.5},
                {"matrix": [5, 14], "x": 15.25, "y": 5.25},
                {"matrix": [5, 15], "x": 16.25, "y": 5.25},
                {"matrix": [5, 16], "x": 17.25, "y": 5.25}
             ]
        }
```

File: keyboards/abatskeyboardclub/newjeans/solder/keymaps/via/keymap.c [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+#include "via.h"
+#include "eeprom_driver.h"
+#include "quantum.h"
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_tkl_f13_ansi_split_bs_rshift(
+        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_F13,              KC_PSCR, KC_SCRL, KC_PAUS,
+        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_BSPC,    KC_INS,  KC_HOME, KC_PGUP,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,             KC_DEL,  KC_END,  KC_PGDN,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
```
Comment (dunk2k) [inline keyboards/abatskeyboardclub/newjeans/solder/keymaps/via/keymap.c:15]:
```suggestion
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_NUHS, KC_ENT,
```

**Review by dunk2k — CHANGES_REQUESTED** [review]
add `LAYOUT_all` layout for solder version, as to conform to PR Checklist

**Comment by dunk2k** [comment]
@ramonimbao 
Do the ISO layouts for the solder version have the usual, for ISO, split (1.25u-1u) Left Shift?

**Review by waffle87 — COMMENTED** [review]
The custom VIA indicator code is currently not accepted as it poses a maintenance burden.

File: keyboards/abatskeyboardclub/newjeans/hotswap/config.h [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+/* VIA custom menu*/
+#define VIA_EEPROM_CUSTOM_CONFIG_SIZE 7
+
+/* RGB */
+#define WS2812_SPI_DRIVER SPID2
+#define WS2812_SPI_MOSI_PAL_MODE 0
+#define WS2812_SPI_SCK_PIN B13
+#define WS2812_SPI_SCK_PAL_MODE 0
+#define WS2812_EXTERNAL_PULLUP
+#define RGB_MATRIX_LED_COUNT 91
```
Comment (dunk2k) [inline keyboards/abatskeyboardclub/newjeans/hotswap/config.h:15]: accounted for with `rgblight.layout` in `info.json`
```suggestion
```

File: keyboards/abatskeyboardclub/newjeans/solder/config.h [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+
+#pragma once
+
+/* RGB */
+#define WS2812_SPI_DRIVER SPID2
+#define WS2812_SPI_MOSI_PAL_MODE 0
+#define WS2812_SPI_SCK_PIN B13
+#define WS2812_SPI_SCK_PAL_MODE 0
+#define WS2812_EXTERNAL_PULLUP
+#define RGB_MATRIX_LED_COUNT 91
```
Comment (dunk2k) [inline keyboards/abatskeyboardclub/newjeans/solder/config.h:12]: Accounted for with `rgblight.layout` in `info.json`
```suggestion
```

**Review by dunk2k — CHANGES_REQUESTED** [review]
RGB_MATRIX_LED_COUNT removed from `config.h`

**Comment by ramonimbao** [comment]
> Do the ISO layouts for the solder version have the usual, for ISO, split (1.25u-1u) Left Shift?

There is no split left shift - only ISO enter is present.

**Comment by ramonimbao** [comment]
> The custom VIA indicator code is currently not accepted as it poses a maintenance burden.

Does this include basic things like controlling LED colors? All I'm doing is grabbing HSV from the custom menus for two LEDs and having that displayed.

**Comment by zvecr** [comment]
> Does this include basic things like controlling LED colors? All I'm doing is grabbing HSV from the custom menus for two LEDs and having that displayed.

Yes it does.

**Comment by ramonimbao** [comment]
Well, Abats has decided not to get this merged anymore and instead we'll maintain it in our own fork. Thanks for taking the time to review despite this, everyone!

**Pull request closed without merging**